src/models/referral_store.rs

```diff
@@ -36,21 +37,30 @@ impl Serialize for AppDataStruct {
         S: Serializer,
     {
         let mut map = serializer.serialize_map(Some(self.app_data.keys().len()))?;
-        for (hash, address) in &self.app_data {
+        for (hash, data) in &self.app_data {
             let mut bytes = [0u8; 2 + 32 * 2];
             bytes[..2].copy_from_slice(b"0x");
             // Can only fail if the buffer size does not match but we know it is correct.
             hex::encode_to_slice(hash, &mut bytes[2..]).unwrap();
             // Hex encoding is always valid utf8.
             let hash_serialized = std::str::from_utf8(&bytes).unwrap();
 
-            if let Referral::Address(Some(address_bytes)) = address {
-                let mut bytes = [0u8; 2 + 20 * 2];
-                bytes[..2].copy_from_slice(b"0x");
-                // Can only fail if the buffer size does not match but we know it is correct.
-                hex::encode_to_slice(address_bytes, &mut bytes[2..]).unwrap();
-                let address_serialized = std::str::from_utf8(&bytes).unwrap();
-                map.serialize_entry(&hash_serialized, &address_serialized.to_string())?
+            if let AppDataJson::Data(Some(app_data)) = data {
+                // TODO - Remove this BEFORE merge
+                // let mut bytes = [0u8; 2 + 20 * 2];
+                // bytes[..2].copy_from_slice(b"0x");
+                // if let Some(referrer) = app_data.read_referrer() {
+                //     let address_bytes = referrer.as_bytes();
+                //     // Can only fail if the buffer size does not match but we know it is correct.
+                //     hex::encode_to_slice(address_bytes, &mut bytes[2..]).unwrap();
+                //     let address_serialized = std::str::from_utf8(&bytes).unwrap();
+                //     map.serialize_entry(&hash_serialized, &address_serialized.to_string())?
+                // }
+                // TODO - serialize entire app_data...
+                // map.serialize_entry(&hash_serialized, &app_data.to_string())?
+                let serialized_app_data = serde_json::to_string(&app_data).unwrap();
+                map.serialize_entry(&hash_serialized, &serialized_app_data)?
```

Comment:
The above code was only ever here to prevent refferer (as H160) to have the shortened format. 
The problem here is that this serialization (inherited from serde_json) gives all these extra backslashes:

```

Left:  {"0x2947be33ebfa25686ec204857135dd1c676f35d6c252eb066fffaf9b493a01b4":"{\"version\":\"1.2.3\",\"appCode\":\"MooSwap\",\"metadata\":{\"environment\":\"production\",\"referrer\":{\"address\":\"0x8c35b7ee520277d14af5f6098835a584c337311b\",\"version\":\" ...

Right: {"0x2947be33ebfa25686ec204857135dd1c676f35d6c252eb066fffaf9b493a01b4":{"appCode":"MooSwap","metadata":{"environment":"production","referrer":{"address":"0x8c35B7eE520277D14af5F6098835A584C337311b","kind":"referrer","version":"6.6.6"}},"version":"1.2.3"}}
<Click to see difference>

thread 'models::referral_store::tests::test_serialization_of_referral' panicked at 'assertion failed: `(left == right)`
  left: `"{\"0x2947be33ebfa25686ec204857135dd1c676f35d6c252eb066fffaf9b493a01b4\":\"{\\\"version\\\":\\\"1.2.3\\\",\\\"appCode\\\":\\\"MooSwap\\\",\\\"metadata\\\":{\\\"environment\\\":\\\"production\\\",\\\"referrer\\\":{\\\"address\\\":\\\"0x8c35b7ee520277d14af5f6098835a584c337311b\\\",\\\"version\\\":\\\"6.6.6\\\"}}}\"}"`,
 right: `"{\"0x2947be33ebfa25686ec204857135dd1c676f35d6c252eb066fffaf9b493a01b4\":{\"appCode\":\"MooSwap\",\"metadata\":{\"environment\":\"production\",\"referrer\":{\"address\":\"0x8c35B7eE520277D14af5F6098835A584C337311b\",\"kind\":\"referrer\",\"version\":\"6.6.6\"}},\"version\":\"1.2.3\"}}"`', src/models/referral_store.rs:119:9
stack backtrace:
```